contribute/design/1.x/upgrade/volume-pools-upgrade.md

```diff
@@ -0,0 +1,298 @@
+conditions are added by the Upgrade Job
+
+```
+apiVersion: openebs.io/v1alpha1
+kind: UpgradeTask
+metadata:
+  name: upgrade-jiva-pv-001
+  namespace: openebs
+spec:
+  fromVersion: 0.9.0
+  toVersion: 1.0.0
+  #flags can be used to change the default behavior 
+  flags: 
+    #maximum seconds to wait at any given step in the upgrade
+    timeout: 0 #wait forever
+  #the resource represents the type of resource being upgraded. 
+  #some examples are jivaVolume, cstorVolume, cstorPool, cstorPoolClaim
+  #and so forth. Each resource may pass a different set of parameters
+  #to uniquely identify the resource being upgraded. 
+  resourceType: 
+    jivaVolume:
+      pvName: pvc-3d290e5f-7ada-11e9-b8a5-54e1ad4a9dd4
+      #flags can be used to change the default behaviour 
+      flags: 
+        #specify the steps that can be ignored on error.
+        #ignoreStepsOnError: 
+        # - PRE_CHECK
+status:
+  phase: #STARTED, SUCCESS or ERROR can be the supported phases
+  #timestamp when the upgrade job started. Set the phase to STARTED
+  startTime: 2019-07-11T17:39:01Z
+  #timestamp when the phase changed to DONE or ERROR
+  completedTime: 2019-07-11T17:40:01Z
+  # upgrade statuses represent the various stages in the upgrade
+  # and the current state of each stage.
+  upgradeDetailedStatuses:
+  #Upgrade can comprise of a series of steps like PRE_UPGRADE, 
+  # TARGET_UPGRADE, REPLICA_UPGRADE, VERIFY, ROLLBACK and so forth
+  # depending on the resource being upgraded.
+  - step:   # Upgrade Stage - PRE_UPGRADE, TARGET_UPGRADE, ...,VERIFY
+    startTime:
+    #state represents the current state of the step. The 
+    #state can be waiting, errored or completed. lastUpdatedAt
+    #will contain the timestamp at which the current state is updated.
+    lastUpdatedAt:
+    state:
+      #The state can be on one of the following 
+      waiting:
+        message: Initiated rollout of "deployment/pvc-dep-name"
+      errored:
+        #extract the error message and reason from kubectl
+        message: Unable to patch "deployment/pvc-dep-name"
+        reason: ErrorRollout
+      completed:
+        #include the details like which resource was patched
+        message: patched "deployment/pvc-dep-name"
+
+```
+
+The upgrade job will pass the name of the upgrade task as ENV. A
+sample upgrade job will look like:
+
+```
+apiVersion: batch/v1
+kind: Job
+metadata:
+  name: upgrade-jiva-pv-001-job
+  namespace: openebs
+spec:
+  template:
+    spec:
+      #If running in openebs, openebs-maya-operator can be used as 
+      #service account.
+      serviceAccountName: super-admin
+      containers:
+      - name:  upgrade
+        image: openebs/upgrade-executor:latest
+        env:
+        - name: UPGRADE_TASK_CR_NAME
+          value: "upgrade-jiva-pv-001"
+        - name: JOB_NAME
+          valueFrom:
+            fieldRef:
+              fieldPath: metadata.name
+        - name: JOB_NAMESPACE
```

Comment:
Can CR and Job have different namespace ?

Reply:
nope. keeping it simple.